src/golang/lib/repos/artifact.go

```diff
@@ -0,0 +1,53 @@
+package repos
+
+import (
+	"context"
+
+	"github.com/aqueducthq/aqueduct/lib/database"
+	"github.com/aqueducthq/aqueduct/lib/models"
+	"github.com/aqueducthq/aqueduct/lib/models/shared"
+	"github.com/google/uuid"
+)
+
+// Artifact defines all of the database operations that can be performed for an Artifact.
+type Artifact interface {
+	artifactReader
+	artifactWriter
+}
+
+type artifactReader interface {
+	// Exists returns whether an Artifact with ID exists.
+	Exists(ctx context.Context, ID uuid.UUID, db database.Database) (bool, error)
```

Comment:
I like the capital `ID` in all the params! I'm going to refactor my changes as well to add this.